### PR TITLE
Add per-casa-org scope and use it in pundit policies

### LINKED ISSUE #1371: Scope per casa org in CaseContactPolicy
There should never be `.all` only `.where(casa_org: @user.casa_org)` in this code and code like it

Add tests to go with this change

```
 def resolve
      case @user
      when CasaAdmin # scope.in_casa_administered_by(@user)
        scope.all
      when Volunteer
        scope.where(casa_case: CasaCase.actively_assigned_to(@user), creator: @user)
      when Supervisor
        scope.all
      else
        raise "unrecognized user type #{@user.type}"
      end
```

### PR DESCRIPTION
**Pull request by gabrielbaldao**
### What github issue is this PR for, if any?
Resolves #1371 

### What changed, and why?
As asking in #1371 I'm changing the scopes to returns just `.where(casa_org: organization)`

### How will this affect user permissions?
- Volunteer permissions:
- Supervisor permissions:
- Admin permissions:

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### CODE REVIEW

File: app/policies/casa_case_policy.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def initialize(user, scope)
     def resolve
       case @user
       when CasaAdmin, Supervisor
-        scope
+        scope.where(casa_org: @user.casa_org)
```
Comment (wenderjean) [inline app/policies/casa_case_policy.rb:20]: Would it make sense given the context to move it into a `scope` in the model?

Comment (compwron) [reply app/policies/casa_case_policy.rb:20]: yes please :)

**Comment by compwron** [comment]
How's it going? I am excited to merge this! It looks like right now there are some spec failures and a small merge conflict. Please let me know here or in slack if you'd like help. And we have office hours every wednesday at 6pm pacific time if you want to come by and ask with voice/video.

**Comment by gabrielbaldao** [comment]
Sorry about it @compwron , are you sure about use where(casa_org:) in every case?
Like, for case contact should be where(casa_case:) right?

**Comment by compwron** [comment]
> Sorry about it @compwron , are you sure about use where(casa_org:) in every case?
> Like, for case contact should be where(casa_case:) right?

@gabrielbaldao I don't understand what your question is. Don't worry about "Would it make sense given the context to move it into a scope in the model?"- let's just get the functionality merged :)

**Review by compwron — APPROVED** [review]
Yes yes yes! Very cool :)
We need more tests around this to make sure we never leak between orgs... 😅

**Pull request merged**